src/main/java/uk/gov/ida/eidas/utils/keyloader/PKCS11KeyLoader.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.ida.eidas.utils.keyloader;
+
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStore.PrivateKeyEntry;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.Provider;
+import java.security.UnrecoverableEntryException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+/**
+ * Partially cribbed from net.shibboleth.tool.xmlsectool.CredentialHelper
+ */
+public class PKCS11KeyLoader {
+  private final Provider provider;
+  private final String keyPassword;
+
+  public PKCS11KeyLoader(final Provider provider, final String keyPassword) {
+    this.provider = provider;
+    this.keyPassword = keyPassword;
+  }
+
+  public PrivateKey getSigningKey(String keyAlias) throws NoSuchAlgorithmException,
+          UnrecoverableEntryException,
+          KeyStoreException,
+          SecurityException,
+          IllegalArgumentException,
+          IOException,
+          CertificateException {
+    final KeyStore keyStore = KeyStore.getInstance("PKCS11", this.provider);
+    keyStore.load(null, keyPassword.toCharArray());
+    final PrivateKeyEntry keyEntry = (PrivateKeyEntry)keyStore.getEntry(keyAlias,
+            new KeyStore.PasswordProtection(keyPassword.toCharArray()));
+    if (keyEntry == null) {
+      throw new KeyStoreException("Key store contains wrong kind of credential, need Private key");
```

Comment:
I've only notices now - this file is indented with 2 spaces but our style guide mandates 4
https://gds-way.cloudapps.digital/manuals/programming-languages/java.html#java-style-guide

Reply:
The whole project is only indented with 2 spaces.

Reply:
Nope - other files are indented with 4. IntelliJ shows indentation in the bottom right-hand corner

Reply:
`SignWithFile` and `SignWithSmartcard` are also indented with 2 spaces - they should be re-indented with 4 to match the rest of the codebase

Reply:
Done